pkg/volume/csi/csi_plugin.go

```diff
@@ -620,10 +631,45 @@ func (p *csiPlugin) skipAttach(driver string) (bool, error) {
 		}
 		return false, err
 	}
-	if csiDriver.Spec.AttachRequired != nil && *csiDriver.Spec.AttachRequired == false {
-		return true, nil
+
+	skipAttach := false
+	if csiDriver.Spec.AttachRequired != nil {
+		skipAttach = !(*csiDriver.Spec.AttachRequired)
+	}
+
+	// TODO (vladimrvivien) include inline volume as part of skip decision
+	/*
+		inlineShouldSkip := (csiDriver.Spec.Lifecycle == string(ephemeralInlineLifecycle)
+		skipAttach = skipAttach || inlineShouldSkip
+	*/
+	return skipAttach, nil
+}
+
+// getInlineLifecycleSupported returns the inline CSIDriver.Spec.Lifecycle supported: {persistent|ephemeral}.
+// This can be interpreted as 1) whether the driver supports inline at all, "" means no inline support
+// 2) CSIDriver.Spec.Lifecycle == "persistent" inline support for persistent volumes only
+// 3) CSIDriver.Spec.Lifecycle == "ephemeral" inline support for ephemeral volumes only
+// See https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190122-csi-inline-volumes.md
+func (p *csiPlugin) getInlineLifecycleSupported(driver string) (inlineLifecycle, error) {
+	// TODO (vladimirvivien) get value directly from CSIDriver object when field is implemented.
+	// For now, only supports persistent lifecycle
+	return persistentInlineLifecycle, nil
+}
+
+// asertVolumeHandle ensures handle is provided using the inline volume flag from CSIDriver.Lifecycle
+// Handle is required: 1) if unable to determine flag value 2) Lifecycle == persistent
+func (p *csiPlugin) assertVolumeHandle(driver, handle string) error {
+	lifecycle, err := p.getInlineLifecycleSupported(driver)
+	if err != nil {
+		klog.Error(log("asserting volumeHandle failed, assuming handle required: %s", err))
+	}
+	if lifecycle == ephemeralInlineLifecycle {
```

Comment:
Since this method is only used for attach related calls, and we don't support attach for ephemeral volumes, can we just have this function validate that a volume handle is specified and that the lifecycle is persistent?